nexus/db-queries/src/db/datastore/external_ip.rs

```diff
@@ -289,4 +289,18 @@ impl DataStore {
             .await
             .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
     }
+
+    /// Fetch all external IP addresses of any kind for the provided instance
+    pub async fn lookup_external_ips(
+        &self,
+        opctx: &OpContext,
+    ) -> LookupResult<Vec<ExternalIp>> {
+        use db::schema::external_ip::dsl;
+        dsl::external_ip
+            .filter(dsl::time_deleted.is_null())
+            .select(ExternalIp::as_select())
+            .get_results_async(self.pool_authorized(opctx).await?)
+            .await
+            .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
+    }
```

Comment:
I think it's pretty dangerous to add a non-authz-gated datastore operation like this.  If this is a generally useful function, I'd just do the authz check.  If we don't think we'd need this kind of functionality in Nexus, or if authz is a problem, I think it's fine to just put this code straight into omdb (without an authz check).

Reply:
Sounds good. I just moved this into the omdb code.